contracts/libraries/Assessment/AssessmentUtilsLib.sol

```diff
@@ -0,0 +1,90 @@
+// SPDX-License-Identifier: GPL-3.0-only
+
+pragma solidity ^0.8.0;
+
+import "../../interfaces/IAssessment.sol";
+import "hardhat/console.sol";
+
+library AssessmentUtilsLib {
+  // Percentages are defined between 0-10000 i.e. double decimal precision
+  uint16 internal constant PERC_BASIS_POINTS = 10000;
+
+  function abs(int x) internal pure returns (int) {
+    return x >= 0 ? x : -x;
+  }
+
+  function min(uint a, uint b) internal pure returns (uint) {
+    return a <= b ? a : b;
+  }
+
+  function pollFraudExists(IAssessment.Poll memory poll) internal pure returns (bool) {
+    return poll.start > 0;
+  }
+
+  // Used in operations involving NXM tokens and divisions
+  uint internal constant PRECISION = 10 ** 18;
+
+  function _getPollStatus(IAssessment.Poll memory poll) internal view returns (IAssessment.PollStatus) {
+    if (block.timestamp < poll.end) {
+      return IAssessment.PollStatus.PENDING;
+    }
+    if (poll.accepted > poll.denied) {
+      return IAssessment.PollStatus.ACCEPTED;
+    }
+    return IAssessment.PollStatus.DENIED;
+  }
+
+  function _getPayoutImpactOfClaim (IAssessment.ClaimDetails memory details)
+  internal pure returns (uint) {
+    return details.amount * PRECISION / details.nxmPriceSnapshot;
+  }
+
+  function _getPayoutImpactOfIncident (IAssessment.IncidentDetails memory details)
+  internal pure returns (uint) {
+    return details.activeCoverAmount * details.impactEstimatePerc / PERC_BASIS_POINTS;
+  }
+
+  function _getVoteLockupEndDate (
+    IAssessment.Configuration calldata CONFIG,
+    IAssessment.Vote memory vote
+   ) internal pure returns (uint) {
+    return vote.timestamp + CONFIG.MAX_VOTING_PERIOD_DAYS + CONFIG.PAYOUT_COOLDOWN_DAYS;
+  }
+
+  function _getCooldownEndDate (
+    IAssessment.Configuration calldata CONFIG,
+    uint32 pollEnd
+  ) internal pure returns (uint32) {
+    return pollEnd + CONFIG.PAYOUT_COOLDOWN_DAYS * 1 days;
+  }
+
+  function _calculatePollEndDate (
+    IAssessment.Configuration calldata CONFIG,
+    uint96 accepted,
+    uint96 denied,
+    uint32 start,
+    uint payoutImpact
+  ) internal pure returns (uint32) {
+    if (accepted == 0 && denied == 0) {
+      return uint32(start + CONFIG.MIN_VOTING_PERIOD_DAYS * 1 days);
+    }
+
```

Comment:
can use some documentation for this function to understand the logic behind it

in the case of tokenDrivenStrength what is the reason to have  '10 * PRECISION' and then divide the whole min(,) by 10?

also not immediately clear why the 2 strengths are the same unit of "measurement" assuming payoutImpact is in NXM